### ISSUE TITLE
UnboundLocalError: local variable 'display' referenced before assignment

### ISSUE THREAD

**Issue by lionlife** [spyder-ide/spyder#3256]
## Description of your problem
When executing this extremely short code, spyder hangs.

**What steps will reproduce the problem?**
[This could be a contest for shortest bug report.]
Here is the program:
____
Str = '\x8c'
print(Str)
____
I then set a breakpoint on line 2 (print line).
I then click "Debug file" button

**What is the expected output? What do you see instead?**
Spyder hangs!

**Please provide any additional information below**
By killing the python process which runs this code, spyder console shows:
continue
> /home/user/Documents/bug_spyder.py(1)<module>()
-> Str = '\x8c'
(Pdb) > /home/user/Documents/bug_spyder.py(2)<module>()
-> print(Str)
(Pdb) Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/dist-packages/spyderlib/widgets/externalshell/monitor.py", line 569, in run
    self.update_remote_view()
  File "/usr/lib/python2.7/dist-packages/spyderlib/widgets/externalshell/monitor.py", line 450, in update_remote_view
    remote_view = make_remote_view(ns, settings, more_excluded_names)
  File "/usr/lib/python2.7/dist-packages/spyderlib/widgets/externalshell/monitor.py", line 79, in make_remote_view
    minmax=settings['minmax'])
  File "/usr/lib/python2.7/dist-packages/spyderlib/widgets/dicteditorutils.py", line 225, in value_to_display
    return display
UnboundLocalError: local variable 'display' referenced before assignment

## Versions and main components

* Spyder Version: Spyder 2.3.8 
* Python Version: Python 2.7.11+ 64bits
* Operating system: Ubuntu 16.04

This issue is not present with spyder3. I only see it with spyder(2).

## Dependencies

Please go to the menu entry `Help > Optional Dependencies` (or
`Help > Dependencies`), press the button `Copy to clipboard`
and paste the contents below:
IPython >=1.0      :  2.4.1 (OK)
jedi >=0.8.1;<0.9.0:  0.9.0 (NOK)
matplotlib >=1.0   :  1.5.1 (OK)
pandas >=0.13.1    :  0.17.1 (OK)
pep8 >=0.6         :  None (NOK)
psutil >=0.3       :  3.4.2 (OK)
pyflakes >=0.5.0   :  1.1.0 (OK)
pygments >=1.6     :  2.1 (OK)
pylint >=0.25      :  1.5.2 (OK)
rope >=0.9.2       :  0.10.2 (OK)
sphinx >=0.6.6     :  1.3.6 (OK)
sympy >=0.7.3      :  None (NOK)
zmq >=2.1.11       :  15.2.0 (OK)

**Comment by lionlife** [comment]
I have found a workaround to avoid spyder from hanging:

```
       elif is_binary_string(value):
            try:
                display = to_text_string(value, 'utf8')
            except:
                pass
```
The pass may be replaced by:
`                display = "<error>"
`

**Comment by ccordoba12** [comment]
I think this was fixed in Spyder 2.3.9. Could you update and try again?

**Comment by ccordoba12** [comment]
Definitely fixed in 2.3.9, with code very similar to what you proposed:

https://github.com/spyder-ide/spyder/blob/2.3/spyderlib/widgets/dicteditorutils.py#L209
